### PR TITLE
ui: Get rid of violating props passed to DOM elements

### LINKED ISSUE #1028: React does not recognize the X prop on a DOM element
In [this commit](https://github.com/eclipse-apoapsis/ort-server/commit/bf74b0b65722321ec37a0b9e40e4038c86ed90a3), `DataTablePagination` was changed to use props instead of table context, to provide type safety for links. This creates a lot of React warnings in dev console for every page:

_Warning: React does not recognize the `X` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `X` instead. If you accidentally passed it from a parent component, remove it from the DOM element._
```
a
Link<@http://localhost:5173/node_modules/.vite/deps/chunk-535XYBFZ.js?v=1d5fd888:3583:35
div
div
DataTablePagination@http://localhost:5173/src/components/data-table/data-table-pagination.tsx:34:36
div
div
DataTable@http://localhost:5173/src/components/data-table/data-table.tsx:31:26
div
```

@mmurto any idea to fix this?  [This seems to be related](https://stackoverflow.com/questions/54468535/how-to-solve-warning-react-does-not-recognize-the-x-prop-on-a-dom-element), and the props in the warnings list router provider's props as well as the `setPageSizeOptions` and `setCurrentPageOptions` of the `DataTablePagination` component. Seems as if the router provider's props are "leaking" down to the pagination component.

### PR DESCRIPTION
**Pull request by Etsija**
Please see the commits for details.

Resolves #1028.

### CODE REVIEW

File: ui/src/components/data-table/data-table.tsx [hunk 1]
```diff
@@ -53,11 +53,15 @@ export function DataTable<TData>({
 }: DataTableProps<TData>) {
   const pagination = table.getState().pagination;
   const totalPages = table.getPageCount();
+  // We need to extract the violating props from the props object
```
Comment (sschuberth) [inline ui/src/components/data-table/data-table.tsx:56]: ORT projects usually follow the convention to use passive voice in comments, commit messages etc. so you shoud better say "Extract violating properties ...". Also in pain text, "props" should be spelled out as "properties" unless it refers directly to the variable name.

File: ui/src/components/data-table/data-table.tsx [hunk 2]
```diff
@@ -53,11 +53,15 @@ export function DataTable<TData>({
...
+  // without using them in the component.
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const { setPageSizeOptions, setCurrentPageOptions, ...restProps } = props;
```
Comment (sschuberth) [inline ui/src/components/data-table/data-table.tsx:59]: As "rest" sounds more like "sleep", maybe better "remainingProps"?

File: ui/src/components/data-table/data-table.tsx [hunk 3]
```diff
@@ -49,16 +49,14 @@ export function DataTable<TData>({
...
 }: DataTableProps<TData>) {
   const pagination = table.getState().pagination;
   const totalPages = table.getPageCount();
 
   return (
-    <div
-      className={cn('w-full space-y-2.5 overflow-auto', className)}
-      {...props}
-    >
+    <div className={cn('w-full space-y-2.5 overflow-auto', className)}>
```
Comment (mmurto) [inline ui/src/components/data-table/data-table.tsx:59]: You should keep the {...props} here and above to allow for all div props to be given to the wrapping div.

**Review by mmurto — CHANGES_REQUESTED** [review]

File: ui/src/routes/_layout/organizations/$orgId/index.tsx [hunk 4]
```diff
@@ -227,13 +227,13 @@ const OrganizationComponent = () => {
             table={table}
             setCurrentPageOptions={(currentPage) => {
               return {
-                ...Route,
+                to: Route.to,
```
Comment (mmurto) [inline ui/src/routes/_layout/organizations/$orgId/index.tsx:230]: Just to make sure, you did try that this works on all pages without passing `params`? I have no doubt that it doesn't work on the organization page at `/`, but does it work for products, repos and runs as well, as those pages do have params?

**Review by mmurto — CHANGES_REQUESTED** [review]

File: ui/src/routes/_layout/organizations/$orgId/index.tsx (hunk 4 above)
Comment (Etsija) [reply ui/src/routes/_layout/organizations/$orgId/index.tsx:230]: Yes, I tried it on all pages, it works without params.

**Review by mmurto — APPROVED** [review]

File: ui/src/components/data-table/data-table.tsx (hunk 1 above)
Comment (sschuberth) [reply ui/src/components/data-table/data-table.tsx]: Ok, then let's stick with that...

**Pull request merged**